lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,538 @@
+      time_to_process: testGranule.timeToPreprocess,
+      product_volume: testGranule.productVolume.toString(),
+      error: testGranule.error,
+      cmr_link: testGranule.cmrLink,
+      execution_cumulus_id: executionCumulusId,
+      pdr_cumulus_id: null,
+      provider_cumulus_id: null,
+      query_fields: null,
+      beginning_date_time: new Date(testGranule.beginningDateTime),
+      ending_date_time: new Date(testGranule.endingDateTime),
+      last_update_date_time: new Date(testGranule.lastUpdateDateTime),
+      processing_end_date_time: new Date(testGranule.processingEndDateTime),
+      processing_start_date_time: new Date(testGranule.processingStartDateTime),
+      production_date_time: new Date(testGranule.productionDateTime),
+      timestamp: new Date(testGranule.timestamp),
+      created_at: new Date(testGranule.createdAt),
+    }
+  );
+});
+
+test.serial('migrateFileRecord correctly migrates file record', async (t) => {
+  const {
+    filePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testFile = testGranule.files[0];
+  await migrateGranuleRecord(testGranule, knex);
+  await migrateFileRecord(testFile, testGranule.granuleId, testGranule.collectionId, knex);
+
+  // I am not sure how I can select a file where bucket and key are null
+  const record = await filePgModel.get(knex, {});
+
+  t.deepEqual(
+    omit(record, fileOmitList),
+    omit({
+      bucket: testFile.bucket ? testFile.bucket : null,
+      checksum_value: testFile.checksum,
+      checksum_type: testFile.checksumType,
+      key: testFile.key ? testFile.key : null,
+      path: null,
+      file_size: testFile.size.toString(),
+      file_name: testFile.fileName,
+      source: testFile.source,
+      type: testFile.type,
+    },
+    ['type'])
+  );
+});
+
+test.serial('migrateGranuleRecord throws error on invalid source data from DynamoDB', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  delete testGranule.collectionId;
+
+  await t.throwsAsync(
+    migrateGranuleRecord(testGranule, knex),
+    { name: 'SchemaValidationError' }
+  );
+});
+
+test.serial('migrateGranuleRecord handles nullable fields on source granule data', async (t) => {
+  const {
+    collectionCumulusId,
+    executionCumulusId,
+    granulePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  delete testGranule.pdrName;
+  delete testGranule.cmrLink;
+  delete testGranule.published;
+  delete testGranule.duration;
+  delete testGranule.files;
+  delete testGranule.error;
+  delete testGranule.productVolume;
+  delete testGranule.timeToPreprocess;
+  delete testGranule.beginningDateTime;
+  delete testGranule.endingDateTime;
+  delete testGranule.processingStartDateTime;
+  delete testGranule.processingEndDateTime;
+  delete testGranule.lastUpdateDateTime;
+  delete testGranule.timeToArchive;
+  delete testGranule.productionDateTime;
+  delete testGranule.timestamp;
+  delete testGranule.provider;
```

Comment:
I think you need to add `version`, `queryFields`, `execution`, and `queryFields` to this list

Reply:
You're right about `version` and `queryFields`. Taking a closer look at the [granules schema ](https://github.com/nasa/cumulus/blob/master/packages/api/models/schemas.js#L362), it looks like execution is required. Is there a reason why it is not required for our PG schema?

Reply:
[`6ad9950` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/6ad99508c7ee725ec8bd8b4c40ca3092192275f4)

Reply:
Good point about `execution`. There's no reason that I can think of not to make it required for PG. Maybe worth asking the team though